tables/basic-table/step7.md

```diff
@@ -0,0 +1,88 @@
+In this next step, we'll add the state properties we need to track how many rows per page the user can display, what the current page the user is on within the dataset, and we'll add a handler to handle updating the number of rows per page when the user changes this.
+
+## Task
+
+1) Create a numPerPage state property, setting its initial value to 2.
+
+<pre class="file" data-target="clipboard">
+const [numPerPage, setNumPerPage] = React.useState(2);
+</pre>
+
+2) Set the `perPage` prop of Pagination to the numPerPage state property you just created.
+
+<strong>Note: </strong> At this point, the Pagination component is aware that we should only be showing 1-2 of 6 items, and that we should be on page 1 of 3. We haven’t updated our rows yet to reflect this. Don't worry, we’ll do that shortly.
+
+3) Create a currentPage state property, setting its initial value to 1. Then, set the `page` prop of the Pagination component to the currentPage state property you just created.
+
+<pre class="file" data-target="clipboard">
+const [numPerPage, setNumPerPage] = React.useState(2);
+</pre>
+
+<strong>Note: </strong> The second parameter in the destructured array coming from React.useState() is the setter for the state variable you're creating.
+
+4) Set the `perPageOptions` property for Pagination so that we have two options in the dropdown, 2 and 3
+
+<pre class="file" data-target="clipboard">
+perPageOptions={[{ title: "2", value: 2 }, { title: "3", value: 3 }]}
+</pre>
+
+5) Set the `onPerPageSelect` property of Pagination to a function called "handlePerPageSelect".
+
+<pre class="file" data-target="clipboard">
+onPerPageSelect={handlePerPageSelect}
+</pre>
+
+6) Create the handler function “handlePerPageSelect” inside your component definition and set the numPerPage property when it is invoked.
+
+<pre class="file" data-target="clipboard">
+const handlePerPageSelect = (event, value) => {
+  // do stuff here
+};
+</pre>
+
+7) Inside the handlePerPageSelect function, update the numPerPage state property with the value that’s passed as the second parameter.
+
+<pre class="file" data-target="clipboard">
+const handlePerPageSelect = (event, value) => {
+  setNumPerPage(value);
+};
+</pre>
```

Comment:
This should be one step. No need to add the "do stuff here' placeholder, only to add the code in the next step.

Should this be created before adding to pagination?